rspace-bench/src/test/scala/coop/rchain/rspace/bench/serialization/ModelSerializerBench.scala

```diff
@@ -0,0 +1,175 @@
+package coop.rchain.rspace.bench.serialization
+
+import java.util.concurrent.TimeUnit
+import java.nio.ByteBuffer
+
+import coop.rchain.models._
+import coop.rchain.models.testImplicits._
+import coop.rchain.rspace.internal._
+import org.scalacheck.Gen.Parameters
+import org.scalacheck.rng.Seed
+import coop.rchain.rspace.bench.serialization._
+
+import org.openjdk.jmh.annotations.{State => BenchState, _}
+import org.openjdk.jmh.infra.Blackhole
+
+class ModelSerializerBench {
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def protobufRoundTrip(bh: Blackhole, state: ProtobufModelBenchState) = {
+    val gnat = state.gnat
+    val res  = state.roundTrip(gnat)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def protobufHeavyRoundTrip(bh: Blackhole, state: ProtobufModelBenchState) = {
+    val res = state.roundTripMany(state.heavyGnats)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def kryoRoundTrip(bh: Blackhole, state: KryoModelBenchState) = {
+    val gnat = state.gnat
+    val res  = state.roundTrip(gnat)(state.serializer)
+    bh.consume(res)
+  }
+
+  @Benchmark
+  @BenchmarkMode(Array(Mode.AverageTime))
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def kryoHeavyRoundTrip(bh: Blackhole, state: KryoModelBenchState) = {
+    val res = state.roundTripMany(state.heavyGnats)(state.serializer)
+    bh.consume(res)
+  }
+
+}
+
+abstract class ModelSerializerBenchState {
+  type TestGNAT = GNAT[Par, BindPattern, ListParWithRandom, TaggedContinuation]
+
+  implicit def serializer: Serialize2ByteBuffer[TestGNAT]
+
+  def roundTrip[A](in: A)(implicit s: Serialize2ByteBuffer[A]): A = {
+    val meta = s.encode(in)
+    val out  = s.decode(meta)
+    assert(out == in)
+    out
+  }
+
+  def roundTripMany[A](seq: Seq[A])(implicit s: Serialize2ByteBuffer[A]): Seq[A] =
+    seq.map(roundTrip(_))
```

Comment:
This is potential GC and processing time overhead. Granted, there might be none - but it's best to minimize moving parts for a credible benchmark.

Instead we should do

```
state.haeavyGnats.foreach {
  bh.consume(state.roundTrip(_))
}
```

in the tests using this method

Reply:
There's also some annotation in JMH allowing to say that 'single execution of this benchmark method should be counted as N ops' - where N = 1000 here. Please use it.